lib/xml_utils.pm

```diff
@@ -51,26 +51,36 @@ Verifies that node by given XPath is unique and has expected value. C<%args> is
 
 =item * C<xpath> - XPath to the node which value we want to check
 
-=item * C<expected_val> - expected value for the node
+=item * C<expected_value> - expected value for the node`
+
+=item * C<search_by_value> - search is performed by value instead of by node, which is default`
 
 =back
 
 =cut
 
 sub verify_option {
     my (%args) = @_;
-
-    my @nodes = find_nodes(%args);
-    ## Verify that there is node found by xpath and it's single one
-    if (scalar @nodes != 1) {
-        return "Generated autoinst.xml contains unexpected number of nodes for xpath: $args{xpath}. Found: " . scalar @nodes . ", expected: 1.";
+    if ($args{search_by_value}) {
+        my $value = $args{xpc}->findvalue($args{xpath});
+        if ($value ne $args{expected_value}) {
+            return "Unexpected value for xpath $args{xpath}. Expected: '$args{expected_value}', got: '$value'";
+        }
     }
-    if ($nodes[0]->to_literal ne $args{expected_val}) {
-        return "Unexpected value for xpath $args{xpath}. Expected: '$args{expected_val}', got: '$nodes[0]'";
+    else {
+        my @nodes = find_nodes(%args);
+        if (scalar @nodes == 0) {
+            return "Expected xpath $args{xpath} didn't return any node";
+        }
```

Comment:
i dont know if we follow some convention here but if not (still can applied though with another way!) the perl way is more readable 
return "Expected xpath $args{xpath} didn't return any node" if (scalar @nodes == 0);

Reply:
No, in this case is not more readable because the following checks are not in postfix way and the message are pretty long, I would need to put all the message in variables which is even worst... Besides that I'm keeping same structure of previous code, not doing this part from scratch.

Reply:
but I agree with you that those `if/else` looks like could be done in a different way, perhaps having some hash for the message, not sure, ...

Reply:
no need to make it complicated. was just a suggestion. resolved